spec/controllers/foods_controller_spec.rb

```diff
@@ -1,159 +1,191 @@
-      }
+    describe 'GET #new' do
+      it "assigns a new food as @food" do
+        get :new
+        expect(assigns(:food)).to be_a_new(Food)
+      end
+    end
 
-      it "updates the requested food" do
-        food = Food.create! valid_attributes
-        put :update, {:id => food.to_param, :food => new_attributes}, valid_session
-        food.reload
-        skip("Add assertions for updated state")
+    describe 'GET #edit' do
+      it "raises an error if a non-user food is requested" do
+        food = FactoryGirl.create(:food)
+        expect { get(:edit, id: food.to_param) }.
+          to raise_error(ActiveRecord::RecordNotFound)
       end
 
       it "assigns the requested food as @food" do
-        food = Food.create! valid_attributes
-        put :update, {:id => food.to_param, :food => valid_attributes}, valid_session
+        food = FactoryGirl.create(:food, user: @current_user)
+        get :edit, id: food.to_param
         expect(assigns(:food)).to eq(food)
       end
+    end
 
-      it "redirects to the food" do
-        food = Food.create! valid_attributes
-        put :update, {:id => food.to_param, :food => valid_attributes}, valid_session
-        expect(response).to redirect_to(food)
+    describe 'POST #create' do
+      context "with valid params" do
+        it "creates a new Food" do
+          expect { post :create, food: valid_attributes }.
+            to change(Food, :count).by(1)
+        end
+
+        it "assigns a newly created food as @food for the user" do
+          post :create, food: valid_attributes
+          expect(assigns(:food)).to be_a(Food)
+          expect(assigns(:food)).to be_persisted
+          expect(assigns(:food).user).to eq(@current_user)
+        end
+
+        it "redirects to the food list if no recipe is present" do
+          post :create, food: valid_attributes
+          expect(response).to redirect_to(foods_url)
+        end
+
+        it "redirects to the recipe if recipe is present" do
+          recipe = FactoryGirl.create(:recipe)
+          post :create, food: valid_attributes.merge(recipe_id: recipe.to_param)
+          expect(response).to(redirect_to(new_recipe_ingredient_url(recipe, ingredient: { food_id: Food.last.to_param })))
+        end
+      end
+
+      context "with invalid params" do
+        it "assigns a newly created but unsaved food as @food" do
+          post :create, food: { name: "" }
+          expect(assigns(:food)).to be_a_new(Food)
+        end
+
+        it 're-renders the "new" template' do
+          post :create, food: { name: "" }
+          expect(response).to render_template("new")
+        end
       end
     end
 
-    context "with invalid params" do
-      it "assigns the food as @food" do
-        food = Food.create! valid_attributes
-        put :update, {:id => food.to_param, :food => invalid_attributes}, valid_session
-        expect(assigns(:food)).to eq(food)
+    describe 'PUT #update' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.